### ISSUE TITLE
Crash while funding a channell --> messed outputs

### ISSUE THREAD

**Issue by mloop1** [ElementsProject/lightning#2398]
### Issue and Steps to Reproduce
After funding a channel and having as response "lightning-cli: reading response: Success" the node crashed.
When trying to restart anche checking the listfunds the ```outputs``` unfortunately disappeared.
It seems the issue was triggered by a database action which resulted in a unique constraint violation.

```+35740.761365546 lightningd(24827):DEBUG: lightning_openingd-0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: peer_in WIRE_ACCEPT_CHANNEL
+35740.764482895 lightning_hsmd(24836):DEBUG: Client: Received message 19 from client
+35740.764638682 lightningd(24827):DEBUG: lightning_openingd-0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: signature 3045022100d05538db57fbbc82575ff186a02b0a91d133394716fc9fe6c7b6b0e173e688b502201988759df9a8b786d5eeb147a179e031259ae6d8f5893a1efae0ad9c13c0bbc501 on tx 0200000001564359b89aab2b3a35c6c4477fa77301b755460951c65b84d7dfae811e8c82340000000000eb12ac80012c61050000000000160014c936a173b669ba970ebbf77375c6532fcecd5514d4fae920 using key 028061977569801d311220cfdbada7cd0dccff29c9b134e3ccfd8d63de6bbdaecf
+35740.764767296 lightningd(24827):DEBUG: lightning_openingd-0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: peer_out WIRE_FUNDING_CREATED
+35741.884988535 lightningd(24827):DEBUG: lightning_openingd-0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: peer_in WIRE_FUNDING_SIGNED
+35741.886903360 lightningd(24827):DEBUG: lightning_openingd-0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: UPDATE WIRE_OPENING_FUNDER_REPLY
+35741.887018993 lightningd(24827):DEBUG: lightning_openingd-0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: UPDATE WIRE_OPENING_FUNDER_REPLY
+35741.887185998 lightningd(24827):DEBUG: 0386ab3ee5f80eb2c0a24bfa7317221b0432636f40a3d24f32e04539c68aad49fc
+35741.890550394 lightningd(24827):DEBUG: 0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: Funding tx has 1 inputs, 2 outputs:
+35741.890608587 lightningd(24827):DEBUG: 0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266 chan #4923: 0: 14147067 satoshi (SEGWIT) 9e99e21ea7d9f07ac73ad34639eb21594a145c0a869cb73c165a9b565646c2e4?
+35741.892014113 lightningd(24827):BROKEN: wallet_channel_insert: UNIQUE constraint failed: peers.node_id
+35741.926342515 lightningd(24827):BROKEN: FATAL SIGNAL 6 (version v0.6.3rc1-116-gf1a837e)
+35741.926376071 lightningd(24827):BROKEN: backtrace: common/daemon.c:45 (crashdump) 0x439358
+35741.926396282 lightningd(24827):BROKEN: backtrace: (null):0 ((null)) 0x7fbd711554af
+35741.926409049 lightningd(24827):BROKEN: backtrace: (null):0 ((null)) 0x7fbd71155428
+35741.926419148 lightningd(24827):BROKEN: backtrace: (null):0 ((null)) 0x7fbd71157029
+35741.926430507 lightningd(24827):BROKEN: backtrace: lightningd/log.c:624 (fatal) 0x41b11d
+35741.926442616 lightningd(24827):BROKEN: backtrace: wallet/db.c:448 (db_exec_prepared_) 0x4620ba
+35741.926461292 lightningd(24827):BROKEN: backtrace: wallet/wallet.c:1049 (wallet_channel_insert) 0x468ee0
+35741.926476922 lightningd(24827):BROKEN: backtrace: lightningd/opening_control.c:218 (wallet_commit_channel) 0x41de43
+35741.926491154 lightningd(24827):BROKEN: backtrace: lightningd/opening_control.c:374 (opening_funder_finished) 0x41e703
+35741.926508928 lightningd(24827):BROKEN: backtrace: lightningd/opening_control.c:692 (openingd_msg) 0x41f42a
+35741.926521970 lightningd(24827):BROKEN: backtrace: lightningd/subd.c:474 (sd_msg_read) 0x4338d2
+35741.926534694 lightningd(24827):BROKEN: backtrace: lightningd/subd.c:302 (read_fds) 0x43322e
+35741.926547084 lightningd(24827):BROKEN: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x47808c
+35741.926559040 lightningd(24827):BROKEN: backtrace: ccan/ccan/io/io.c:395 (do_plan) 0x478b81
+35741.926571178 lightningd(24827):BROKEN: backtrace: ccan/ccan/io/io.c:405 (io_ready) 0x478bbf
+35741.926582857 lightningd(24827):BROKEN: backtrace: ccan/ccan/io/poll.c:310 (io_loop) 0x47a71f
+35741.926595179 lightningd(24827):BROKEN: backtrace: lightningd/lightningd.c:827 (main) 0x419254
+35741.926607291 lightningd(24827):BROKEN: backtrace: (null):0 ((null)) 0x7fbd7114082f
+35741.926620355 lightningd(24827):BROKEN: backtrace: (null):0 ((null)) 0x4038d8
+35741.926629554 lightningd(24827):BROKEN: backtrace: (null):0 ((null)) 0xffffffffffffffff
```

### `getinfo` output
  "version": "v0.6.3rc1-116-gf1a837e", 
  "blockheight": 564503, 
  "network": "bitcoin",

**Comment by cdecker** [comment]
The root cause is fixed in latest `master`, and will soon be released.

As for your outputs you can call `dev-rescan-outputs` and they should be reset to the actual state.

**Comment by cdecker** [comment]
For reference this is the fix: https://github.com/ElementsProject/lightning/pull/2371

**Comment by mloop1** [comment]
thanks @cdecker .
I updated to v0.70rc1 and the rescan fixed the outputs... I enjoy being reckless :)

Tried to re-fund the channel again and got this message:
{ "code" : -1, "message" : "They sent error channel 16750cdbbbcf104a72e552d18e1606edb640124b8017191655c3c486a04bd301: 01" }

**Comment by yaraaer** [comment]
5 days ago, I funded new channel with low fee. I waited 5 days for confirmation. lnc wallet had single unconfirmed UTXO (change) and I couldn't open any new channel. To work around that, I transferred a million sat to lnc wallet from an exchange. I got 1 unconfirmed and 1 confirmed utxo in the end.
now.. I tried to fund additional new channel, but each time I executed fundchannel, error "lightning-cli: reading response: Success" occured and lnc crashed. I tried to open new channels with 3 different nodes, every time it crashed. today, my unconfirmed utxo is finally confirmed, so I tried to open the very same new channels - success. (end of story)
attaching example of crash.log
[crash.log](https://github.com/ElementsProject/lightning/files/3195230/crash.log)
